flutter/apps/prompting_client_ui/linux/my_application.cc

```diff
@@ -56,10 +95,37 @@ static void my_application_activate(GApplication* application) {
   gtk_container_add(GTK_CONTAINER(window), GTK_WIDGET(view));
 
   fl_register_plugins(FL_PLUGIN_REGISTRY(view));
+
+  // Extract the first two arguments that don't have leading dashes from dart_entrypoint_arguments.
+  char *snap_name = NULL;
+  guint64 app_id = 0;
+  for (uint i = 0; self->dart_entrypoint_arguments[i] != NULL; i++) {
+    char *arg = self->dart_entrypoint_arguments[i];
+    if (*arg == '-') {
+      continue;
+    }
+
+    if (snap_name == NULL) {
+      snap_name = arg;
+      continue;
+    }
+
+    app_id = g_ascii_strtoull(arg, NULL, 10);
+    if (errno != 0 || app_id == 0) {
+      app_id = 0;
+      g_warning("failed to get PID from the application prompted for: %s is not a valid uint64", arg);
+    }
+
+    break;
+  }
+  
+  signal_prompting_to_gnome_shell(snap_name, app_id);
```

Comment:
If we end up here, we should not even try to signal this to the shell, but very likely to exit the application IMHO